### DISCUSSION TITLE
Resources for learning V as a first language

### DISCUSSION THREAD

**Discussion by chrisdebian** [vlang/v#20535] [General]
Posted in Discord, but I suspect others may ask a similar question in the future, so it may be good to have a record, here.

"Hi, all. My first post, I've only known about 'V', for an hour! I really like the philosophy and goals of the language. I am new to programming; is V too new and immature for a beginner to learn? If it isn't, could anyone point me to any good resources, for a complete beginner? I like books, but I'm guessing we're not at that stage, yet. Cheers, Chris"

**Comment by JalonSolov** [comment]
From the [Wikipedia page](https://en.wikipedia.org/wiki/V_(programming_language))

Further reading
* The V Programming Language basic (in Japanese). Independent Laboratory. 20 June 2020. [ASIN](https://en.wikipedia.org/wiki/ASIN_(identifier)) [B08BKJDRFR](https://www.amazon.com/dp/B08BKJDRFR).
* Rao, Navule Pavan Kumar (10 December 2021). Getting Started with V Programming. [Packt Publishing](https://en.wikipedia.org/wiki/Packt_Publishing).
[ISBN](https://en.wikipedia.org/wiki/ISBN_(identifier)) [978-1839213434](https://en.wikipedia.org/wiki/Special:BookSources/978-1839213434). [OCLC](https://en.wikipedia.org/wiki/OCLC_(identifier)) [1290492862](https://www.worldcat.org/oclc/1290492862).
* Lyons, Dakota "Kai" (13 April 2022). Beginning with V Programming. Independently Published. [ISBN](https://en.wikipedia.org/wiki/ISBN_(identifier)) [979-8801499963](https://en.wikipedia.org/wiki/Special:BookSources/979-8801499963).
* Chakraborty, Dr. Soubhik; Haldar, Subhomoy (6 December 2023). Randomness Revisited using the V Programming Language. Nova Science Publishers. [doi](https://en.wikipedia.org/wiki/Doi_(identifier)):[10.52305/CVCN5241](https://doi.org/10.52305%2FCVCN5241). [ISBN](https://en.wikipedia.org/wiki/ISBN_(identifier)) [979-8891133280](https://en.wikipedia.org/wiki/Special:BookSources/979-8891133280).
* Tsoukalos, Mihalis (May 2022). "Discover the V language". Linux Format Magazine (288). [ISSN](https://en.wikipedia.org/wiki/ISSN_(identifier)) [1470-4234](https://www.worldcat.org/issn/1470-4234).

2 of the authors hang out in the V discord, as well... Navule Pavan Kumar Rao (@navule on discord) and Subhomy Haldar (@hungrybluedev).

**Comment by chrisdebian** [comment]
Brilliant, that's really helpful; I'll have a look at the links.

Chris

**Comment by Wajinn** [comment]
There are also videos, that can help get into and familiarize with the subject:

**1)** [An introduction to V - the vlang](https://www.youtube.com/watch?v=ZO3vlwqvTrc) 
**2)** [A small presentation of V's features at IBM](https://www.youtube.com/watch?v=_dIAcNp9bJs)
**3)** [How To Maintain And Iterate With V - SYNCS 2023](https://www.youtube.com/watch?v=pchEsYwA74Q)

**Reply by JalonSolov** [reply]
Should get those up on the Wikipedia page.

**Comment by enghitalo** [comment]
> There are also videos, that can help get into and familiarize with the subject:
> 
> **1)** [An introduction to V - the vlang](https://www.youtube.com/watch?v=ZO3vlwqvTrc) 
> **2)** [A small presentation of V's features at IBM](https://www.youtube.com/watch?v=_dIAcNp9bJs)
> **3)** [How To Maintain And Iterate With V - SYNCS 2023](https://www.youtube.com/watch?v=pchEsYwA74Q)

https://www.youtube.com/playlist?list=PLpz113t3_XTuEF9jpDxvvWatK9BLBbb6W

**Comment by EchoPouet** [comment]
You have exercices in [exercism](https://exercism.org/tracks/vlang). It's free.

**Reply by chrisdebian** [reply]
Thanks you, I didn't know about exercism.

Chris
